plugins/inputs/ravendb/ravendb_test.go

```diff
@@ -199,8 +192,23 @@ func TestRavenDBGeneratesMetricsFull(t *testing.T) {
 		"public_server_url": "http://localhost:8080",
 	}
 
-	compareFields(t, collectionFields, acc, "ravendb_collections")
-	compareTags(t, collectionTags, acc, "ravendb_collections")
+	serverExpected := testutil.MustMetric("ravendb_server", serverTags, serverFields, defaultTime)
+	dbExpected := testutil.MustMetric("ravendb_databases", dbTags, dbFields, defaultTime)
+	indexExpected := testutil.MustMetric("ravendb_indexes", indexTags, indexFields, defaultTime)
+	collectionsExpected := testutil.MustMetric("ravendb_collections", collectionTags, collectionFields, defaultTime)
+
+	for _, metric := range acc.GetTelegrafMetrics() {
+		switch metric.Name() {
+		case "ravendb_server":
+			testutil.RequireMetricEqual(t, serverExpected, metric, testutil.IgnoreTime())
+		case "ravendb_databases":
+			testutil.RequireMetricEqual(t, dbExpected, metric, testutil.IgnoreTime())
+		case "ravendb_indexes":
+			testutil.RequireMetricEqual(t, indexExpected, metric, testutil.IgnoreTime())
+		case "ravendb_collections":
+			testutil.RequireMetricEqual(t, collectionsExpected, metric, testutil.IgnoreTime())
+		}
+	}
```

Comment:
If I'm not mistaken, you can add the expected metrics to an `[]telegraf.Metric` array and then test all metrics at once with `testutil.RequireMetricsEqual()` (note the **s** in Metric**s**)!

Reply:
actually I can't use that as all operations take place in parallel and order is not deterministic. That's the reason I have this switch.

Reply:
Well I think `testutil.RequireMetricsEqual()` cares for the ordering. However to avoid race conditions you should call `acc.Wait(N)` before comparison, where `N` is the number of metrics you expect.

Reply:
it won't help as we gather stats and push to accumulator in parallel, see: https://github.com/ml054/telegraf/blob/master/plugins/inputs/ravendb/ravendb.go#L111